### ISSUE TITLE
New 1.19 feature to better group certificates with differences in name fails for "UE / U" in standardized name

### ISSUE THREAD

**Issue by jkrwdf** [Digitaler-Impfnachweis/covpass-android#198]
## Avoid duplicates
* [X] Bug is not mentioned in the [FAQ](https://digitaler-impfnachweis-app.de/faq/)
* [X] Bug is not already reported in another issue

## Technical details

- Device name: Google Pixel 3 XL
- OS version: 12
- App version: 1.19.0

## Describe the bug

In the release notes for 1.19, the following advertisement was made: 

> NEUE FUNKTIONEN
> Die CovPass-App als auch die CovPassCheck-App enthalten nun eine Erklärung zur Barrierefreiheit. Sie gibt Aufschluss darüber, wie barrierefrei die App aktuell ist und was noch verbessert wird. Außerdem kann die App nun eingescannte Zertifikate besser zu den richtigen Personen zuordnen, auch wenn Unterschiede bezüglich Schreibweise des Namens, Titel oder Zweitnamen bestehen.

I could test this today with a RAT certificate where upon issuing, the German Umlaut in my first name "ü" was accidentally entered for the standardized name (and only there) as single "u". 

Assuming my name was "Günther Maier", the standardized name in my vaccination certificates is correctly MAIER<<GUENTHER, while the RAT certificate contains MAIER<<GUNTHER. 

This simplest type of accidental difference is not detected by CovPass, 

I get two distinct cards, and to make it even more confusing for the user, both show the non-standardized name "Günther Maier" identically.

## Steps to reproduce the issue

See above.

## Expected behaviour

See above.

**Comment by Ein-Tim** [comment]
@thomasaugsten Is this something you considered for version 2.19 of CWA?

**Comment by jkrwdf** [comment]
Side remarkl This type of divergence is provoked by frontend software in the test center which requires manual input of the "Standardized First and Last Name" by the personnel for each user and each test. 

I already stumbled across this effect in https://github.com/corona-warn-app/cwa-app-android/issues/3670, when the operators in my center just entered "something" to be able to continue. 

Since then, the instructions were refined, but still this frontend does not deduce the standardized names from the data provided by the registration (which in my case happens by scanning a CWA Rapid Test Profile QR-Code).

**Comment by pucherd** [comment]
Hello @jkrwdf, thank you for reporting this issue. But in this case I think the issue is that the transliterated names (givenName and familyName) are not the same and thus interpreted as no-match -> two different persons.
The input parameters we are using for matching are birthdate, givenNameTransliterated (gnt), familyNameTransliterated (fnt) - if these parameters matches between your QR Codes they are grouped into one card.

**Comment by jkrwdf** [comment]
I understood the announcement for version 1.19 (which appeared in the public as "Now CovPass solves a big problem") as a kind of more relaxed approach, maybe based on a "similarity score" or alike, therefore I had to consider the current behavior as bug, as it seems not to be different from before.

**Comment by jkrwdf** [comment]
Yes, I understand that. 

I still try to find this feature: "Außerdem kann die App nun eingescannte Zertifikate besser zu den richtigen Personen zuordnen, auch wenn Unterschiede bezüglich Schreibweise des Namens, Titel oder Zweitnamen bestehen.".

My case seems not covered here. 
If you can confirm this and consider my case not relevant for this feature, the issue can be closed.

**Comment by timokoenig** [comment]
Regarding `Außerdem kann die App nun eingescannte Zertifikate besser zu den richtigen Personen zuordnen, auch wenn Unterschiede bezüglich Schreibweise des Namens, Titel oder Zweitnamen bestehen.` I haven't seen any improvements in the code except the removal of `Dr.`

**Comment by Ein-Tim** [comment]
@timokoenig 

I can confirm that the solution works, certificates are grouped together despite having a second name in them, etc.

**Comment by alexcimander** [comment]
About the functionality:

To have a "match" for both certificates there must be at least one match within the standardized last name and at least one match within the standardized first name of a certificate (+ same date of birth). Also "DR" will be removed.

E.G:

_Erika Maria Meier_ and _Erika Meier_ would be the same person

_Dr. Erika Maria Meier_ and _Dr. Erika Meier_ would be the same person

but not _Erika Maria Muller_ and _Erika Maria Mueller_ since there not a single match in the last name. 

Having said that there is an edge case for false-positives: Having Twins with the same second name. 
The twins would be grouped "to one person".

So @jkrwdf sadly your case would not be covered.

Does that clarify the functionality? :-)

**Comment by timokoenig** [comment]
It clarifies it, but I would've loved to see those edge cases to be integrated because I am also one and it's not very complicated. My last name has three different writings

KÖNIG
KONIG
KOENIG

**Comment by jkrwdf** [comment]
@alexcimander Thanks for the explanation. 

I agree that the case I described, slight differences in the content of the name parts, is a different thing as a comparison which of those name parts are present or not. 

For me, closing this issue is fine. 

@timokoenig : Would this be OK for you? 

@Ein-Tim : I just got CWA 2.19.0 for Android, and also there, GUNTER and GUENTHER are two separate entities. In the "Neue Funktionen" description for this CWA version, the feature is described as "Kleine Unterschiede in der Schreibweise von Vor- und Nachnamen werden ignoriert". I think I will file a new issue for CWA with my showcase example, because this text definitely promises more than I actually get.
